### ISSUE TITLE
FinCoT prompting (arXiv 2506.16123): Tier-0 prompt-only probe of direction collapse

### ISSUE THREAD

**Issue by dgonier** [dgonier/consciousness-ecology-codd#11]
## Status
PROPOSED

## Motivation
FinCoT (Nitarach et al., arXiv 2506.16123, June 2025 / Feb 2026) is a structured chain-of-thought prompting framework with hand-built domain-expert reasoning blueprints across 10 CFA-style finance domains. **No fine-tuning required.** Three properties make it directly relevant to issue #8:

1. **Works on Qwen3-8B-Base**, takes accuracy 63.2% → 80.5% on CFA-style finance tasks. Qwen3-4B is the closest sibling architecture; the blueprint should translate.
2. **Targets reasoning structure, not domain knowledge.** Matches our diagnosis that direction collapse is a *reasoning* failure dressed as a *prediction* failure.
3. **Authors note: "FinCoT proves most effective for models lacking financial post-training."** That's literally us pre-#10.

This is a Tier-0 intervention — a prompt-engineering probe before we commit to the training-time interventions in #10. If structured prompting alone moves StockNet MCC off zero, we get the architectural lift essentially free.

## Naming collision warning

There are TWO things called "FinCoT":
- **FinCoT (dataset)** = `TheFinAI/Fino1_Reasoning_Path_FinQA`, from Fino1 paper (arXiv 2502.08127, Feb 2025), distilled GPT-4o reasoning traces over FinQA. Already in our #10 plan.
- **FinCoT (framework)** = THIS issue. Hand-built CFA-domain reasoning blueprints used as prompting structure. Nitarach et al., arXiv 2506.16123.

Not duplicates — these compose. The framework's blueprints become input data for SFT on the dataset. We test the framework alone first; if it works, we don't need the dataset training. If it doesn't, we still benefit from the blueprints when training the Interrogator on the dataset.

## Proposal

Two cheap experiments, both ~30-60 min total (no training):

### Experiment A: Predator-side structured-CoT prompting

Modify the predator's `role_prefix` in `trophic/agents/predator.py` to embed a CFA-style structured reasoning blueprint. Something like:

```
You are a short-horizon market predictor following the CFA framework:
1. IDENTIFY the ticker and time horizon from the input.
2. ENUMERATE evidence (price, fundamental, social) found in the input.
3. WEIGH each evidence type by reliability for this scenario.
4. INFER the direction (up/down/hold) consistent with the weighted evidence.
5. CALIBRATE confidence based on evidence agreement.
6. EMIT the prediction in the required schema.

Now produce: ...
```

Then re-run StockNet smoke on `ipo_seed8_best.pt` (the post-#6-fix checkpoint, our current best). Pre-fix smoke had MCC=0. **If MCC > 0.05, we have a paper-grade result without any retraining.**

### Experiment B: Interrogator-gate the predator output

Same idea but apply at the Interrogator herbivore tier. The Interrogator already has a "plan → solve → synthesize" pattern; replace its synthesis prompt with a CFA blueprint and route predator output through it. Tests whether reasoning *gating* (not just reasoning *prompting*) breaks the collapse.

## Why this isn't "we already tried prompting"

Our existing role prefixes are unstructured (one paragraph of role description, no enforced reasoning steps). The paper's structured blueprints are explicitly numbered, domain-keyed, and force the model through intermediate slots before committing to an output. The paper measures an 8.9× output-length reduction *with higher accuracy* — meaning structured prompts produce *more focused* reasoning, not just more reasoning.

## Context pointers
- Files: `trophic/agents/predator.py` (ROLE_PROMPTS dict, lines ~41-54)
- Files: `trophic/agents/herbivore.py` (ROLE_PROMPTS, lines ~50-63)
- Files: `trophic/agents/interrogator_herbivore.py` (the gate target for Experiment B)
- Related issue: #8 (direction collapse — current bottleneck this would address)
- Related issue: #10 (per-species CPT — this is the Tier-0 probe before #10 training kicks in)
- External: arXiv 2506.16123, "FinCoT: Grounding Chain-of-Thought in Expert Financial Reasoning"

## Dependencies
- Blocked by: nothing (prompt-only change; no training, no GPU contention with seed 9 v2)
- Blocks: deciding the priority of #10 (if FinCoT-prompt-only fixes #8, #10 is paper-strengthening not paper-blocking)
- Independent of: #1, #2, #3, #9

## Trigger conditions
Start NOW — Experiment A in parallel with the in-flight SFT seed 9 v2 (no GPU conflict; this run only needs the predator decode for StockNet, not training). Experiment B follows if A is partial.

## Decision criteria
- Experiment A: StockNet MCC on `ipo_seed8_best.pt` with structured role_prefix vs MCC=0 baseline. Target: >0.05 to be interesting; >0.10 to be paper-grade.
- Experiment B: same MCC criterion, on the Interrogator-gated predator output.
- Output length: should match or beat the paper's 8.9× reduction (sanity check the framework is being applied correctly).

## Open questions
- Which CFA domain blueprint best matches our task (short-horizon directional prediction with magnitude/horizon/sigma fields)? The paper has 10 domains; "Quantitative Methods" is the closest.
- Does the structured prompt break the existing XML output schema, or do they compose? The paper's outputs are free-form CFA-style answers; we need XML.
- Does it interact with the existing role_prefix.mean() Q construction (#6 fix)? The new prompt is longer, so role_prefix.mean() shifts.

## Effort estimate
- Experiment A: ~1 hour (read paper appendix for blueprint examples, draft a CFA-style prompt for the predator, re-run StockNet smoke).
- Experiment B: ~1-2 hours (modify Interrogator's synthesizer prompt, test end-to-end).
- Plus reading the paper's appendix carefully to use the actual blueprints, not my interpretation.

## Out of scope
- Training-time integration (that's #10)
- Multi-language CFA domains
- The full 10-domain coverage — pick the best-fit domain first

**Comment by dgonier** [comment]
## Experiment A drafted, queued for execution after seed 9 v2 completes

Read the FinCoT paper and pulled the actual blueprint. Key adaptation: the paper's blueprints are CFA-domain-keyed (10 domains); for our short-horizon directional prediction the closest fits are:
- **Equity Investments** (Buy/Hold/Sell decision framework)
- **Quantitative Methods** (parameter estimation + hypothesis testing discipline)

Drafted a hybrid prompt that combines both as a 6-step procedure (IDENTIFY → ENUMERATE → WEIGH → INFER → CALIBRATE → EMIT) preserving the existing XML output schema.

Script lives at `scripts/diagnostics/exp_A_fincot_prompt.py` — idle on disk, monkey-patches `ROLE_PROMPTS['short_horizon']` BEFORE Predator.ensure_initialized fires (the role_prefix is encoded once at agent init via `host.encode_role_prefix`, so the patch order matters).

## Why running now would conflict
SFT seed 9 v2 is using the GPU (PID 52754, 11min in, step 150, eval=0.834 — fast pace now that the eval-token-cap fix landed). Running the experiment in parallel risks OOM. Will run after seed 9 v2 finishes; that's also a clean separation — experiment A specifically tests `ipo_seed8_best.pt` (the prior best), not seed 9.

## Decision criteria preserved
- MCC > 0.05 — interesting; structural reasoning helps without training
- MCC > 0.10 — paper-grade; one-line architecture intervention solves #8
- MCC ≈ 0.0  — falsifies prompting-only hypothesis; route to #10 training fix

## Source citation in script
Adapted from arXiv 2506.16123 page 15-16 (Equity Investments + Quantitative Methods blueprints). Original prompts are 'You are a CFA expert' style with mermaid-graph procedural diagrams; my version preserves the procedural step structure while keeping the existing 'You are a short-horizon market predictor' role identity and the XML output schema.

Will run + report MCC after seed 9 v2 cascade finishes.

**Comment by dgonier** [comment]
## Experiment A failed to test the hypothesis (not the same as falsifying it)

The 1201-char FinCoT prompt swap on `ipo_seed8_best.pt` produced **pure gibberish output**, not abstention as the eval harness reported.

Sample raw decoded text on AAPL_2015-10-01 (target=down):
```
PREDi  p r e d Ie THER E R. SPOILER: P REDI CERE DETERmining horser, herer HER HERS ER HER Her er Er HeR HErErHerHeR HER hER HER HER HER_H
```

The eval parser counts unparseable output as "abstention" → 50/50 fake-abstentions reported.

## Why this happened

The trained policy was conditioned on the original **604-character role prefix**. Swapping in a 1201-character prompt **doubles the prefix length**, which:

1. Shifts `role_prefix.mean(dim=0)` (the role component of the #6-fix hunter_state Q) far out of the trained distribution
2. Trained Channel weights expect the original Q geometry
3. Cross-attention output collapses
4. Predator decode produces character-level garbage

Verified via `scripts/diagnostics/exp_A_inspect.py` running with max_new_tokens=400 — output is nonsense across all token positions.

## What this means

Experiment A failed to *test* the hypothesis it was designed to test. The trained policy can't survive any inference-time prompt change of this magnitude — that's a different finding (architectural fragility) than "FinCoT prompting doesn't break direction collapse."

## Honest verdict on the FinCoT hypothesis

**Inconclusive at inference time.** Not testable without retraining.

Two paths:
- Retrain a new SFT seed with the FinCoT structured prompt baked in from the start — then we'd be testing the hypothesis honestly. But this is essentially what #10 (per-species CPT, Interrogator-on-FinCoT-dataset) proposes already, just with a bigger training set.
- Try a SHORTER structured-CoT prompt that's closer in length to the original 604 chars. Less informative test (the paper used full multi-paragraph blueprints) but might survive the trained policy.

## What we learned (not nothing)

The architecture has a previously-undocumented fragility: **role_prefix length shifts shift the Q distribution and break trained Channel weights**. This is a real bug worth opening as its own issue. It also means our prompts are essentially frozen at training time — any future role-prefix experiments need to be done by retraining, not by inference-time swaps.

Closing this issue as **status:declined** (the hypothesis can't be tested cheaply on existing checkpoints; the cheap-test idea was the whole motivation). Routing forward via #10 and a new fragility issue.
